localstack-core/localstack/services/s3/resource_providers/aws_s3_bucket.py

```diff
@@ -604,7 +604,8 @@ def _create_bucket_if_does_not_exist(self, model, region_name, s3_client):
         try:
             s3_client.head_bucket(Bucket=model["BucketName"])
         except ClientError as e:
-            if e.response["Error"]["Message"] != "Not Found":
+            # TODO: check why the ClientError is not the same, something with Boto
+            if e.response["ResponseMetadata"]["HTTPStatusCode"] != 404:
```

Comment:
Did you find out why this is necessary?

Reply:
Not exactly, but I believe `HeadBucket` is a bit problematic, I remember having some issue in an S3 test with it not really raising the right error too. I could try to look more in detail again why this happened 🤔